### DISCUSSION TITLE
SendPhoto by id

### DISCUSSION THREAD

**Discussion by HighlandProger** [rubenlagus/TelegramBots#1122] [Ideas]
Hi, dear developers!

I'm using your library for a long time. And now i cannot find the oportunity to send photo by it's id on telegram or URL. Your function receives only InputFile objects. Please add the feature to receive String object or explain me how to use the same way to send photo by it's id. I also tried to use your's example bot: https://telegram.me/filesbot (Send files by file_id). It doesn't answer.

King regards,
Java developer
Said Kimpaev
magashdaha@mail.ru

**Comment by NaveenB2004** [comment]
Hello!

InputFile object (org.telegram.telegrambots.meta.api.objects.InputFile) allows you to set **string values** and **files (java.io.File)**. Please see the examples.

_by link or file id:_
```
SendPhoto photo = new SendPhoto();
photo.setChatId("chat id here!");
photo.setPhoto(new InputFile("link or file_id"));
execute(photo);
```

_by file (java.io.File):_
```
SendPhoto photo = new SendPhoto();
photo.setChatId("chat id here!");
photo.setPhoto(new InputFile(new File("file path here!")));
execute(photo);
```

Hope this will help you!

**Reply by abdumajid2001** [reply]
thank you . did it very good
